### ISSUE TITLE
how to finetune different style dataset together and use them in the same dialogue process?

### ISSUE THREAD

**Issue by andysun-sw** [facebookresearch/ParlAI#4546]
Two datasets: 
1 generate poem with keywords
2 blended_skill_talk

After finetuning, the system behave well in each scene. But when I wanted to  merge two abilitys together, it failed badly:
![image](https://user-images.githubusercontent.com/89009298/168508725-1aa21975-e58e-4588-b793-6ec824897021.png)

my training process is as following:
parlai train_model -t blended_skill_talk,wizard_of_wikipedia,convai2:normalized,empathetic_dialogues --multitask-weights 1,3,3,3 -veps 0.1 --attention-dropout 0.0 --batchsize 16 --model transformer/generator --embedding-size 2560 --ffn-size 10240 --variant prelayernorm --n-heads 32 --n-positions 128 --n-encoder-layers 2 --n-decoder-layers 24 --history-add-global-end-token end --delimiter '  ' --dict-tokenizer bytelevelbpe  --dropout 0.1 --fp16 True --init-model zoo:blender/reddit_3B/model --dict-file zoo:blender/reddit_3B/model.dict --label-truncate 128 --log_every_n_secs 10 -lr 7e-06 --lr-scheduler reduceonplateau --lr-scheduler-patience 3 --optimizer adam --relu-dropout 0.0 --activation gelu --model-parallel true --save-after-valid True --text-truncate 128 --truncate 128 --warmup_updates 100 --fp16-impl mem_efficient --update-freq 2 --gradient-clip 0.1 --skip-generation False -vp 10 -vmt ppl -vmm min --beam-block-ngram 3 --model-file

**Comment by stephenroller** [comment]
This is a classic multitasking result. Its too easy for the model to tell the two datasets apart. You either need some sort of smaller, truly hybrid dataset, or you need to massage both datasets to be more similar.

**Comment by andysun-sw** [comment]
you mean that there is no difference if I just put two different style dataset  into one dataset?
I need to creat a new dataset which has two abilities at first?
